File: internal/astgrep/testdata/fixtures/ruby/violation.rb
# Ruby violation fixture — demonstrates patterns matched by ruby rules.
# This file should produce >= 1 finding when scanned with the ruby rule set.

# Matches ruby-unused-var: variable assigned nil
x = nil

# Matches ruby-nil-method-call: calling length on nil
bad_len = nil.length

# Matches ruby-todo-marker: stub implementation
def not_implemented
  raise "TODO"
end

File: internal/astgrep/testdata/fixtures/csharp/violation.cs
// C# violation fixture — demonstrates patterns matched by csharp rules.
// This file should produce >= 1 finding when scanned with the csharp rule set.

using System;

public class Example
{
    public void BadCode()
    {
        // Matches csharp-unused-var: string variable assigned null
        string x = null;

        // Matches csharp-null-deref: Convert.ToString with null argument
        var s = Convert.ToString(null);

        Console.WriteLine(x);
        Console.WriteLine(s);
    }

    // Matches csharp-todo-marker: NotImplementedException stub
    public string NotImplemented()
    {
        throw new NotImplementedException();
    }
}
